tests/headers/noreturn.h

```diff
@@ -0,0 +1,3 @@
+_Noreturn void f(void);
+// TODO (pvdrz): figure out how to handle this case.
+__attribute__((noreturn)) void g(void);
```

Comment:
We have `has_attribute` already, see how we detect `warn_unused_result`, doesn't that work?

Reply:
Sadly no because the `_Noreturn` token has kind `CXToken_Keyword` and `has_attr` only checks for `CXToken_Identifier`. I could change `has_attr` and add a extra argument `token_kind` so we can keep a single function for both.